### PR TITLE
imapdedup: new port

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 11.5.1 20G80 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: mail/imapdedup/Portfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        quentinsf IMAPdedup ec56dbdc687c73808336fc38098c268a0177e57a
+
```
Comment (herbygillot) [inline mail/imapdedup/Portfile:8]: This looks good. Let's add a `revision` field and it should be good to go:
```
github.setup        quentinsf IMAPdedup ec...
revision            0
```

**Comment by kencu** [comment]
I already use this script, but I guess I got it directly... works great. We have imapsync so just makes sense.

File: mail/imapdedup/Portfile (hunk 1 above)
Comment (catap) [reply mail/imapdedup/Portfile:8]: it's done!

File: mail/imapdedup/Portfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+github.setup        quentinsf IMAPdedup ec56dbdc687c73808336fc38098c268a0177e57a
+revision            0
+name                imapdedup
+
+version             20210813
+
+categories          mail
+license             GPL-2
+platforms           darwin
```
Comment (mf2k) [inline mail/imapdedup/Portfile:15]: Since this is a python script. Add the following line here:
`supported_archs     noarch
`

**Review by mf2k — CHANGES_REQUESTED** [review]

File: mail/imapdedup/Portfile (hunk 2 above)
Comment (catap) [reply mail/imapdedup/Portfile:15]: @mf2k it's done!

**Review by herbygillot — APPROVED** [review]

**Review by mf2k — APPROVED** [review]

**Pull request merged**